app/src/main/java/be/digitalia/mediasession2mqtt/MainWorker.kt

```diff
@@ -79,12 +84,70 @@ class MainWorker @Inject constructor(
         }
     }
 
+    private suspend fun publishHassConfiguration(
+        client: MQTTPublishClient,
+        qosLevel: MQTTQoSLevel,
+        deviceId: Int
+    ) {
+        // Using as a unique identifier, but this may not be available on all devices.
+        val serialNumber = Build.SERIAL
+
+        @Serializable
+        data class DeviceInfo(
+            @EncodeDefault val name: String = "MediaSession2MQTT",
+            @EncodeDefault val manufacturer: String = Build.MANUFACTURER,
+            @EncodeDefault val model: String = Build.MODEL,
+            // Passing serialNumber in as a variable causes build fail: "Exception during IR lowering"
+            @EncodeDefault val identifiers: List<String> = listOf(Build.SERIAL),
+            @EncodeDefault val serial_number: String = Build.SERIAL
+        )
+
+        @Serializable
+        data class Sensor(
+            val name: String,
+            val state_topic: String,
+            val unique_id: String,
+            val device: DeviceInfo
+        )
+
+        // TODO: using a data class for storing sensor metadata would be nicer and let us configure
```

Comment:
I can understand not wanting more extra dependencies, somehow I missed `JsonWriter` while searching for how to perform json encoding in kotlin :-) 

I can definitely change that, but I'm wondering if it'd change the situation if the playback information would also be send as a json object in a single topic instead separate topics per bare value? This way it'd be easily extendable, in case there exists other information that could be useful to expose in the future.

Reply:
Is it a good practice to regroup multiple topics in a single Json entry for MQTT? It looks to me that it's more efficient to only update the value that actually changed.
I don't really mind the dependency (it's quite small) but I think in this case object mapping doesn't add much value and doesn't make the code much simpler.

Reply:
To be honest, I'm not sure, but most of the projects I know about (homeassistant, zigbee2mqtt, ..) are using JSON payloads, probably because it's easier to consume and extend if ever needed. This way the updates are also atomic so the state on client's side is always consistent.